src/coreclr/jit/emitarm64.cpp

```diff
@@ -13941,8 +14132,16 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
         case IF_SVE_AC_3A: // ........xx...... ...gggmmmmmddddd -- SVE integer divide vectors (predicated)
         case IF_SVE_AD_3A: // ........xx...... ...gggmmmmmddddd -- SVE integer min/max/difference (predicated)
         case IF_SVE_AE_3A: // ........xx...... ...gggmmmmmddddd -- SVE integer multiply vectors (predicated)
+        case IF_SVE_AF_3A: // ........xx...... ...gggnnnnnddddd -- SVE bitwise logical reduction (predicated)
```

Comment:
I am actually thinking that instead of adding all the SVE instructions and format in existing switch case, we should have them handled under `default` to minimize the TP impact. Something like:

```c++
switch (fmt)
{
  case non_sve_fmt1:
     break;
  case non_sve_fmt2:
     break;
  ...
 default:
    emitOutputSveInstr(...);
    break;
}

emitOutputSveInstr()
{
  switch(..)
 {
   ...
   default:
     assert(!"unexpected");
     break;
 }
}
```

From #95127, here is the impact:

<img width="462" alt="image" src="https://github.com/dotnet/runtime/assets/12488060/41d80101-d333-4b41-ae53-962051fd8015">

Reply:
I'm happy with that.

Addition thoughts:
- Add an assert in the default case: `assert(fmt >= first_sve_fmt && fmt <= last_sve_fmt)`.
- Do this for all the functions with SVE checks. Eg `Emit_R_R_R()`, `emitOutputInstr()` `getInsExecutionCharacteristics()` etc etc.
- Move all the new SVE functions to a new file emitarm64sve.cpp

Were you planning on doing that?

Reply:
> I'm happy with that.
> 
> Addition thoughts:
> 
> * Add an assert in the default case: `assert(fmt >= first_sve_fmt && fmt <= last_sve_fmt)`.
> * Do this for all the functions with SVE checks. Eg `Emit_R_R_R()`, `emitOutputInstr()` `getInsExecutionCharacteristics()` etc etc.
> * Move all the new SVE functions to a new file emitarm64sve.cpp

They all sound good to me.

> Were you planning on doing that?

I was not (yet). It just that I checked the TP impact that made me realize that we should do it. Let me know if you want to do it as part of this PR or I am happy to do it but this PR will have to wait for it to finish. Let me know what works for you.

Reply:
> I was not (yet). It just that I checked the TP impact that made me realize that we should do it. Let me know if you want to do it as part of this PR or I am happy to do it but this PR will have to wait for it to finish. Let me know what works for you.

Happy to hold this PR back. In the meantime, I'll probably add a few more into it. Either way, I didn't want to start something new myself today.

Reply:
latest runs from this PR makes me wonder how much to trust them:

<img width="551" alt="image" src="https://github.com/dotnet/runtime/assets/12488060/414f0078-0018-4e28-9572-ee5034500ed7">

but I guess moving the sve code in `emitarm64sve.cpp` will be general goodness.

Reply:
If moving sve out _does_ improve throughput, then you might want to consider doing the same for NEON instructions? This might make NEON throughput slower, but would improve throughput for general instructions.

NEON is only used for longer running, frequently used methods, so the compiler throughput is less important. Whereas short running, single use methods will all be general instructions, so the compiler throughput matters more for those.

(it also shrinks file size, which makes me happy)